### PR TITLE
For #1312, #1236, #1237, #1238, #1239: Creating, Editing, and Deleting

### LINKED ISSUE #1312: FNX-4951 ⁃ Edit Bookmark
As a user, I want to be able to edit my saved bookmark

### Acceptance Criteria (Added by PM)
I can edit the name of a saved bookmark.
I can edit the URL of a saved bookmark.

### PR DESCRIPTION
**Pull request by colintheshots**
… Bookmarks and Bookmark Folders

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -323,24 +323,30 @@ class BrowserFragment : Fragment(), BackHandler {
...
                         val session = requireComponents.core.sessionManager.selectedSession
                         CoroutineScope(IO).launch {
-                            requireComponents.core.bookmarksStorage
+                            val guid = requireComponents.core.bookmarksStorage
                                 .addItem(BookmarkRoot.Mobile.id, session!!.url, session.title, null)
                             launch(Main) {
                                 val rootView =
                                     context?.asActivity()?.window?.decorView?.findViewById<View>(android.R.id.content)
-                                rootView?.let { view ->
+                                val snackbar = rootView?.let { view ->
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: Jeff just made a branded FenixSnackbar that I'm assuming we could use here?

File: app/src/main/res/layout/component_sign_in.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ekager) [inline app/src/main/res/layout/component_sign_in.xml]: nit - license

File: app/src/main/res/layout/fragment_add_bookmark_folder.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ekager) [inline app/src/main/res/layout/fragment_add_bookmark_folder.xml]: nit - license

File: app/src/main/res/layout/fragment_edit_bookmark.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ekager) [inline app/src/main/res/layout/fragment_edit_bookmark.xml]: nit - license

File: app/src/main/res/layout/fragment_select_bookmark_folder.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ekager) [inline app/src/main/res/layout/fragment_select_bookmark_folder.xml]: license

File: app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -73,6 +73,10 @@
         <item name="historyURLColor">@color/history_url_normal_theme</item>
         <item name="historyHeader">@color/history_header_normal_theme</item>
 
+        <!-- Bookmark fragment colors -->
+        <item name="bookmarksLabelColor">@color/bookmarks_label_normal_theme</item>
```
Comment (ekager) [inline app/src/main/res/values/styles.xml]: I think these colors @color/bookmarks_label_normal_theme and @color/bookmarks_edit_normal_theme should also be overloaded in values-night/colors/xml for dark theme

**Comment by sblatz** [comment]
Do we really want to "close" all of these issues? I think we should likely label them "QA needed" and just have this PR reference them with a "For #ISSUE_NUMBER"

**Pull request merged**